src/RedisIPCMessenger.ts

```diff
@@ -110,6 +110,10 @@ export default class RedisIPCMessenger implements IPCMessenger {
         throw new Error(`Failed to map ${key} to a callback`);
       }
 
+      if (instance === this.instance) {
```

Comment:
Why don't we move this closer to the  top of the block, to be able to short-circuit?

Reply:
both checks can abort the handling, now that the destructuring is replaced by `parts`  array, it makes sense to process the inidividual parts in the order they appear in the array